### ISSUE TITLE
getLine1Number - Missing optional permissions

### ISSUE THREAD

**Issue by galvanu** [react-native-community/react-native-device-info#268]
Hi,

We got crash on Android devices due to missing READ_PHONE_STATE permissions when trying to get device phone number.
READ_SMS permission is granted but for some reason the app still crash.

The exception is thrown from this line in RNDeviceModule.java:
`constants.put("phoneNumber", telMgr.getLine1Number());`

**Happens on Android versions**: 5.x
**React Native version**: 0.48.4
**react-native-device-info version**: 0.12.1
**AndroidManifestPermissions**:
   ```
 <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
    <uses-permission android:name="android.permission.READ_SMS" />
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.READ_PHONE_STATE" tools:node="remove"/>
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
```

**StackTrace**:
```
Java exception in 'NativeModules'

java.lang.SecurityException: Neither user 10058 nor current process has android.permission.READ_PHONE_STATE.
android.os.Parcel.readException
    Parcel.java:1540
android.os.Parcel.readException
    Parcel.java:1493
com.android.internal.telephony.ITelephony$Stub$Proxy.getLine1NumberForDisplay
    ITelephony.java:3247
android.telephony.TelephonyManager.getLine1NumberForSubscriber
    TelephonyManager.java:1720
android.telephony.TelephonyManager.getLine1Number
    TelephonyManager.java:1704
com.learnium.RNDeviceInfo.RNDeviceModule.getConstants
    RNDeviceModule.java:169
com.facebook.react.bridge.JavaModuleWrapper.getConstants
    JavaModuleWrapper.java:140
android.os.Handler.handleCallback
    Handler.java:739
android.os.Handler.dispatchMessage
    Handler.java:95
com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage
    MessageQueueThreadHandler.java:31
<unknown>
    deviceinfo.js:5:57
loadModuleImplementation
    require.js:174:12
guardedLoadModule
    require.js:126:36
_require
    require.js:110:20
<unknown>
    logSender.js:3
loadModuleImplementation
    require.js:174:12
guardedLoadModule
    require.js:126:36
_require
    require.js:110:20
<unknown>
    App.js:5
loadModuleImplementation
    require.js:174:12
guardedLoadModule
    require.js:126:36
_require
    require.js:110:20
<unknown>
    index.android.js:5
loadModuleImplementation
    require.js:174:12
guardedLoadModule
    require.js:119:45
_require
    require.js:110:20
global code
   

```

Any idea why I'm getting this exception even if READ_SMS permission is granted?

**Comment by galvanu** [comment]
After deeper investigation for this issue looks like the required permissions have changed with Android versions:

For Lollipop (the version I have this issues) looks like the getLine1Number method requires READ_PHONE_STATE permissions - [source code](https://android.googlesource.com/platform/frameworks/base.git/+/lollipop-release/telephony/java/android/telephony/TelephonyManager.java)

For newer versions, Marshmallow and higher, google added optional READ_SMS permission for this method - [source code](https://android.googlesource.com/platform/frameworks/base.git/+/marshmallow-release/telephony/java/android/telephony/TelephonyManager.java)

Am I missing something or should the module check first what Android versions is running and ask check for permission according to Android version. Do you want me to create a pull request for this?

**Comment by galvanu** [comment]
Creating a pull request for this #269

**Comment by ewall-cauc** [comment]
thx for your hints, it helps me to get phonenumber

**Comment by felansu** [comment]
@galvanu thanks

**Comment by mikehardy** [comment]
closed with #269
